src/Microsoft.VisualStudio.Editors/Common/Utils.vb

```diff
@@ -1665,5 +1666,29 @@ Namespace Microsoft.VisualStudio.Editors.Common
 
         End Class
 #End Region
+
+        Public Class SerializationProvider
```

Comment:
Rather than duplicate this source, your options are either:

- Add an `InternalsVisibleTo` (seems overkill just for this).
- Make it a top-level internal type in its own source file, and link the file into both projects.

I'd recommend the second approach.

Reply:
There are multiple functions/subroutines in each Utils files that are duplicates. I even created one when doing the BinaryFormatter telemetry. I've been *really* wanting to clean up this code. If I end up making a shared source file, I'd clean up this whole file (Utils) to work in either project; unless there are types specific to a particular project. I think it would make sense to do that in a separate PR, though. I also need to make another PR to remove the BinaryFormatter telemetry since it won't be as necessary after this PR.

Reply:
Created an issue for this here: https://github.com/dotnet/project-system/issues/7351